### PR TITLE
mgr/dashboard: Use $localize and locl for TS translations

### PR DESCRIPTION
**Pull request by tspmelo**
## Checklist
- [x] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard backend

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts [hunk 1]
```diff
@@ -155,33 +153,30 @@ export class HostsComponent extends ListWithDetails implements OnInit {
...
-                  empty: this.i18n('There are no labels.'),
-                  filter: this.i18n('Filter or add labels'),
-                  add: this.i18n('Add label')
-                },
-                this.i18n
-              )
+              messages: new SelectMessages({
+                empty: $localize`There are no labels.`,
+                filter: $localize`Filter or add labels`,
+                add: $localize`Add labe`
```
Comment (bk201) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts:169]:
```suggestion
                add: $localize`Add label`
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts [hunk 2]
```diff
@@ -155,33 +153,30 @@ export class HostsComponent extends ListWithDetails implements OnInit {
         return { enabled: true, name: label };
       });
       this.modalService.show(FormModalComponent, {
-        titleText: this.i18n('Edit Host: {{hostname}}', host),
+        titleText: $localize`Edit Host: ${host}`,
```
Comment (bk201) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts:156]:
```suggestion
        titleText: $localize`Edit Host: ${host.hostname}`,
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts [hunk 3]
```diff
@@ -155,33 +153,30 @@ export class HostsComponent extends ListWithDetails implements OnInit {
...
           }
         ],
-        submitButtonText: this.i18n('Edit Host'),
+        submitButtonText: $localize`Edit Host`,
         onSubmit: (values: any) => {
           this.hostService.update(host['hostname'], values.labels).subscribe(() => {
             this.notificationService.show(
               NotificationType.success,
-              this.i18n('Updated Host "{{hostname}}"', host)
+              $localize`Updated Host "${host}"`
```
Comment (bk201) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts:179]:
```suggestion
              $localize`Updated Host "${host.hostname}"`
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/inventory/inventory-devices/inventory-devices.component.ts [hunk 4]
```diff
@@ -196,24 +194,21 @@ export class InventoryDevicesComponent implements OnInit, OnDestroy {
...
+      submitButtonText: $localize`Execute`,
       onSubmit: (values: any) => {
         this.orchService.identifyDevice(hostname, device, values.duration).subscribe(() => {
           this.notificationService.show(
             NotificationType.success,
-            this.i18n(`Identifying '{{device}}' started on host '{{hostname}}'`, {
-              hostname,
-              device
-            })
+            $localize`Identifying '{{device}}' started on host '{{hostname}}'`
```
Comment (bk201) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/inventory/inventory-devices/inventory-devices.component.ts:211]:
```suggestion
            $localize`Identifying '${device}' started on host '${hostname}'`
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts (hunk 2 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/hosts/hosts.component.ts]: good catch. I will have to double check if this isn't happening on other places...

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/smart-list.component.ts [hunk 5]
```diff
@@ -57,16 +52,12 @@ export class SmartListComponent implements OnInit, OnChanges {
       if (this.isSmartError(smartData)) {
         let userMessage = '';
         if (smartData.smartctl_error_code === -22) {
-          userMessage = this.i18n(
-            `Smartctl has received an unknown argument (error code {{code}}). \
+          userMessage = $localize`Smartctl has received an unknown argument \
+(error code {{smartData.smartctl_error_code}}). \
```
Comment (bk201) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/shared/smart-list/smart-list.component.ts:56]:
```suggestion
(error code ${smartData.smartctl_error_code}). \
```

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
jenkins test dashboard backend

**Pull request merged**